akka-typed/src/main/scala/akka/typed/internal/EventStreamImpl.scala

```diff
@@ -0,0 +1,332 @@
+/**
+ * Copyright (C) 2016 Lightbend Inc. <http://www.lightbend.com/>
+ */
+package akka.typed
+package internal
+
+import akka.{ actor ⇒ a, event ⇒ e }
+import java.util.concurrent.atomic.AtomicReference
+import scala.annotation.tailrec
+import scala.concurrent.{ Await, Promise }
+import akka.util.{ ReentrantGuard, Subclassification, SubclassifiedIndex }
+import scala.collection.immutable
+import java.util.concurrent.TimeoutException
+import akka.util.Timeout
+
+/**
+ * An Akka EventStream is a pub-sub stream of events both system and user generated,
+ * where subscribers are ActorRefs and the channels are Classes and Events are any java.lang.Object.
+ * EventStreams employ SubchannelClassification, which means that if you listen to a Class,
+ * you'll receive any message that is of that type or a subtype.
+ *
+ * The debug flag in the constructor toggles if operations on this EventStream should also be published
+ * as Debug-Events
+ */
+private[typed] class EventStreamImpl(private val debug: Boolean)(implicit private val timeout: Timeout) extends EventStream {
```

Comment:
Unfortunately the untyped implementation is not really modular, which means that due to the changed type signatures the whole implementation needs to be redone as well. Instead of spreading it out over many polymorphic traits it is compactly collected into a single file here.